signer/src/transaction_coordinator.rs

```diff
@@ -827,6 +857,71 @@ where
         }
     }
 
+    fn authenticate_message(
```

Comment:
We could refactor this to be used in both the coordinator and the signer, so that we have a single shared fn doing the authentication logic. WDYT?

Reply:
Agreed @djordon , it's a nice idea but it might be more effort than it's worth.  Also we do things in the various match branches, so we'd have to have another match statement if we moved this code into a shared fn.